pkg/restapi/webhook/webhook.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webhook
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+const notificationInterval = 2 * time.Second
+
+var logger = log.New("aries-framework/agentd")
+
+type dispatcher struct {
+	client         *http.Client
+	subscriberURLs []string
+}
+
+// StartWebhookDispatcher will launch the webhook dispatcher in a new goroutine.
+// The dispatcher is responsible for sending notifications to subscribers.
+func StartWebhookDispatcher(subscriberURLs []string) {
+	dispatcher := &dispatcher{
+		client:         &http.Client{},
+		subscriberURLs: subscriberURLs,
+	}
+
+	go dispatcher.start()
+}
+
+func (d *dispatcher) start() {
+	ticker := time.NewTicker(notificationInterval)
+	defer ticker.Stop()
+	for range ticker.C {
+		d.dispatch()
+	}
+}
+
+func (d *dispatcher) dispatch() {
+	for _, subscriberURL := range d.subscriberURLs {
+		go func(subscriberURL string) {
```

Comment:
Rather than running a go routine per webhook url, it's better to have just one go routine to dispatch all messages to all the webhook urls. The ticker from `start()` should be moved into this go routine's loop.

Reply:
We should not spinup a goroutine for each.

Reply:
Done!